quic/s2n-quic-core/src/random.rs

```diff
@@ -23,7 +23,7 @@ pub trait Generator: 'static + Send {
 /// NOTE: This will have slight bias towards the lower end of the range. Usages that
 /// require uniform sampling should implement rejection sampling or other methodologies
 /// and not copy this implementation.
-pub(crate) fn gen_range_biased<R: Generator>(
+pub(crate) fn gen_range_biased<R: Generator + ?Sized>(
```

Comment:
why is `?Sized` needed?

Reply:
[This post](https://github.com/pretzelhammer/rust-blog/blob/master/posts/sizedness-in-rust.md#sized-in-generics) does a good job of explaining it but TL;DR generics automatically add `+ Sized` unless you opt out of it with `?Sized`. When using `dyn Trait`, its size isn't known at compile time so any calls to a function with `+ Sized` will complain.